x-pack/platform/plugins/shared/onechat/public/application/context/send_message/send_message_context.tsx

```diff
@@ -18,11 +19,20 @@ interface SendMessageState {
   canCancel: boolean;
   cancel: () => void;
   cleanConversation: () => void;
+  connectorSelection: {
+    connectors: ReturnType<typeof useConnectorSelection>['connectors'];
+    selectedConnector: string | undefined;
+    selectConnector: (connectorId: string) => void;
+    isLoading: boolean;
+    defaultConnectorId?: string;
+  };
 }
 
 const SendMessageContext = createContext<SendMessageState | null>(null);
 
 export const SendMessageProvider = ({ children }: { children: React.ReactNode }) => {
+  const connectorSelection = useConnectorSelection();
```

Comment:
Not sure how I feel about mixing the connector selection hook / state directly in that send_message state. OTOH that state is already storing too many things (or the name is no longer reflecting what it's doing at least), so it's probably fine...

Reply:
yeah feels like Vercel's `useChat` at this poiint